src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/RazorPagesRazorViewEngineOptionsSetup.cs

```diff
@@ -10,29 +11,45 @@ namespace Microsoft.AspNetCore.Mvc.RazorPages.Internal
 {
     public class RazorPagesRazorViewEngineOptionsSetup : IConfigureOptions<RazorViewEngineOptions>
     {
-        private readonly IOptions<RazorPagesOptions> _pagesOptions;
+        private readonly RazorPagesOptions _pagesOptions;
 
         public RazorPagesRazorViewEngineOptionsSetup(IOptions<RazorPagesOptions> pagesOptions)
         {
-            _pagesOptions = pagesOptions;
+            _pagesOptions = pagesOptions?.Value ?? throw new ArgumentNullException(nameof(pagesOptions));
         }
 
         public void Configure(RazorViewEngineOptions options)
         {
-            Debug.Assert(_pagesOptions.Value.RootDirectory.Length > 0);
-
-            if (_pagesOptions.Value.RootDirectory == "/")
-            {
-                options.PageViewLocationFormats.Add("/{1}/{0}" + RazorViewEngine.ViewExtension);
-            }
-            else
+            if (options == null)
             {
-                options.PageViewLocationFormats.Add(_pagesOptions.Value.RootDirectory + "/{1}/{0}" + RazorViewEngine.ViewExtension);
+                throw new ArgumentNullException(nameof(options));
             }
 
+            var rootDirectory = _pagesOptions.RootDirectory;
+            Debug.Assert(!string.IsNullOrEmpty(rootDirectory));
```

Comment:
These shouldn't be asserts, they are clearly possible.

Reply:
We disallow these in the property setter (https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.RazorPages/RazorPagesOptions.cs#L32), but sure, we can throw for extra goodness.

Reply:
Oh ok, unimportant then